sympy/physics/control/lti.py

```diff
@@ -0,0 +1,655 @@
+    >>> tf7**3
+    TransferFunction((a + s)**3, (-a + s)**3, s)
+    >>> tf7**0
+    TransferFunction(1, 1, s)
+    >>> tf8 = TransferFunction(p + 4, p - 3, p)
+    >>> tf8**-1
+    TransferFunction(p - 3, p + 4, p)
+
+    References
+    ==========
+
+    .. [1] http://www.cds.caltech.edu/~murray/amwiki/index.php?title=First_Edition
+
+    """
+    def __new__(cls, num, den, var):
+        num, den = _sympify(num), _sympify(den)
+
+        if not isinstance(var, Symbol):
+            raise TypeError("Variable input must be a Symbol.")
+        if den == 0:
+            raise ValueError("TransferFunction can't have a zero denominator.")
+
+        if (((isinstance(num, Expr) and num.has(Symbol) and not num.has(exp)) or num.is_number) and
+            ((isinstance(den, Expr) and den.has(Symbol) and not den.has(exp)) or den.is_number)):
+                obj = super(TransferFunction, cls).__new__(cls, num, den, var)
+                obj._num = num
+                obj._den = den
+                obj._var = var
+                return obj
+        else:
+            raise TypeError("Unsupported type for numerator or denominator of TransferFunction.")
+
+    @property
+    def num(self):
+        """
+        Returns the numerator polynomial/expression of the transfer function.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> G1 = TransferFunction(s**2 + p*s + 3, s - 4, s)
+        >>> G1.num
+        p*s + s**2 + 3
+        >>> G2 = TransferFunction((p + 5)*(p - 3), (p - 3)*(p + 1), p)
+        >>> G2.num
+        (p - 3)*(p + 5)
+
+        """
+        return self._num
+
+    @property
+    def den(self):
+        """
+        Returns the denominator polynomial/expression of the transfer function.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> G1 = TransferFunction(s + 4, p**3 - 2*p + 4, s)
+        >>> G1.den
+        p**3 - 2*p + 4
+        >>> G2 = TransferFunction(3, 4, s)
+        >>> G2.den
+        4
+
+        """
+        return self._den
+
+    @property
+    def var(self):
+        """
+        Returns the variable used to anchor the expressions/polynomials of
+        the transfer function.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> G1 = TransferFunction(p**2 + 2*p + 4, p - 6, p)
+        >>> G1.var
+        p
+        >>> G2 = TransferFunction(0, s - 5, s)
+        >>> G2.var
+        s
+
+        """
+        return self._var
+
+    def _eval_subs(self, old, new):
+        arg_num = self.num.subs(old, new)
+        arg_den = self.den.subs(old, new)
+        argnew = TransferFunction(arg_num, arg_den, self.var)
+        return self if old == self.var else argnew
```

Comment:
Do you have a test for this clause?

Reply:
Yes, [here](https://github.com/sympy/sympy/blob/d6771df476f1c2e30010d5fc38f7c88a6671f297/sympy/physics/control/tests/test_lti.py#L193).